pkg/image/api/validation/validation.go

```diff
@@ -185,3 +183,45 @@ func ValidateImageStreamTagUpdate(newIST, oldIST *api.ImageStreamTag) fielderror
 
 	return result
 }
+
+func ValidateImageStreamImport(isi *api.ImageStreamImport) fielderrors.ValidationErrorList {
```

Comment:
I don't think you should allow a non-nil `spec.repository` and `len(spec.images)` at the same time.